src/ai_migrate/storage/config_loader.py

```diff
@@ -0,0 +1,178 @@
+import os
+import yaml
+from pathlib import Path
+from typing import Optional, Dict, Union
+
+from .config import StorageConfig
+
+DEFAULT_CONFIG_NAME = ".ai-migrate.yml"
+PROJECT_MARKERS = [
+    "pyproject.toml",
+    "setup.py",
+    "setup.cfg",
+    ".ai-migrate-root",
+    DEFAULT_CONFIG_NAME,
+]
+
+DEFAULT_CONFIG_CONTENT = {"type": "local", "path": "migrations"}
+
+
+def find_project_root(start_path: Union[str, Path] = None) -> Optional[Path]:
+    """Find the project root by looking for project marker files.
+
+    @param start_path Starting path for the search (default: current directory)
+    @return Path to project root or None if not found
+    """
+    current = Path(start_path or os.getcwd()).resolve()
+
+    while current != current.parent:
+        # Check for any of the marker files
+        for marker in PROJECT_MARKERS:
+            if (current / marker).exists():
+                return current
+        current = current.parent
+
+    return None
+
+
+def find_config_file(directory: Path) -> Optional[Path]:
+    """Find configuration file in the given directory.
+
+    @param directory Directory to search in
+    @return Path to config file or None if not found
+    """
+    config_path = directory / DEFAULT_CONFIG_NAME
+    return config_path if config_path.exists() else None
+
+
+def create_default_config(path: Path) -> Path:
+    """Create default configuration file.
+
+    @param path Path where to create the config file
+    @return Path to created config file
+    """
+    config_path = path / DEFAULT_CONFIG_NAME
+    with open(config_path, "w") as f:
+        yaml.safe_dump(DEFAULT_CONFIG_CONTENT, f, default_flow_style=False)
+
+    # Also create root marker if it doesn't exist
+    root_marker = path / ".ai-migrate-root"
+    if not root_marker.exists():
+        root_marker.touch()
+
+    return config_path
+
+
+def load_config_file(path: Union[str, Path]) -> Dict[str, Union[Dict, str]]:
+    """Load configuration from a file.
+
+    @param path Path to configuration file
+    @return Configuration dictionary
+    @throws ValueError if file cannot be loaded
+    """
+    path = Path(path).expanduser()
+    if not path.exists():
+        return {}
```

Comment:
since you're passing a path, I'd expect a FileNotFound here

Reply:
good point